test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceTests.cs

```diff
@@ -49,5 +49,29 @@ public void AsSourceItem_WorksCorrectly()
 
             SettingsTestUtils.DeepEquals(source.AsSourceItem(), expectedItem).Should().BeTrue();
         }
+
+        [Fact]
```

Comment:
Could we add more tests for `http://my.test/v3/index.json`, `localhost` to prevent from regression?